### PR TITLE
refs #19 - check for existence of input source asap and fail fast

### PR DESCRIPTION
**Pull request by thatcher**
refs #19 - check for existence of input source asap and fail fast with a... clear message.  no reason to go beyond that point since pypandoc will also fail with a much less clear message

Sorry Jan, not sure how to attach a merge request directly to a ticket.  I even borked my rebase somehow and just recreated my clone.  I guess I've been away for from github for a couple years and a little rough on git skills (was never that good to start). Hopefully this is a clean PR now.

### CODE REVIEW

File: knitpy/knitpy.py [hunk 1]
```diff
@@ -706,6 +706,12 @@ def render(self, filename, output=None):
         filename = os.path.abspath(filename)
         self.log.info("Converting %s..." % filename)
 
+        # no reason to continue past this point if we dont have
+        # a valid file path, so check it here and provide a clear
+        # failure message since this a common mistake
+        if not os.path.exists(filename):
```
Comment (JanSchulz) [inline knitpy/knitpy.py]: Can you have a look how that error message looks like when you are "just" in a wrong directory? As it it after the `abspath`, it has some mangling and will probably be funny for someone who has just used a filename?

**Comment by JanSchulz** [comment]
A (new file with) a unittests would be nice, but I guess I should look at myself first for that :-)

**Comment by thatcher** [comment]
I will do.  In fact, I'm going to close this pull request, apologies.  I really should be branching for each pull request so the pull request can be isolated better.  Its all coming back to be, just trying to 'git' my sea legs back.

**Comment by JanSchulz** [comment]
git clone https://github.com/JanSchulz/knitpy.git
cd knitpy
git remote add mine git@github.com:thatcher/knitpy.git # replace with your clone
git checkout origin/master
git branch -b "fix_for_x"
# changes...
git add knitpy\whatever.py
git commit
git push mine # there will be a message to set the default for the brach -> use that comand again...
# -> in github in your clone do a PR
# if something is wrong, add further commits and
git push mine
# if something is wroing with your last commit
# reedit file
git add file
git commit --amend
git push -f
# or rebase on to of the latest upstream
git pull origin
git rebase origin/master
or if you want to reorder the commits or whateer :-)
git rebase origin/master -i

**Pull request closed without merging**